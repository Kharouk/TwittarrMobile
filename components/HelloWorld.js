import React, { Component } from "react";
import { StyleSheet, Text, View, TextInput, Button } from "react-native";

export default class HelloWorld extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: "Placeholder",
      display: false
    };
  }

  showInformation = () => {
    this.setState({ text: "ðŸ¦Ÿ", display: true });
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Send Message:</Text>
        <TextInput
          value={this.state.text}
          onChangeText={text => this.setState({ text })}
        />
        <Button title="Send" onPress={this.showInformation} />
        {this.state.display == true && <Text>Message Sent.</Text>}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF"
  },
  welcome: {
    fontSize: 40,
    textAlign: "center",
    margin: 10
  }
});
