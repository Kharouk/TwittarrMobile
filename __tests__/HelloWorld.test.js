import React from "react";
import HelloWorld from "../components/HelloWorld";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

test("renders correctly", () => {
  const comp = renderer.create(<HelloWorld />).toJSON();
  expect(comp).toMatchSnapshot();
});

describe("HelloWorld  component", () => {
  it("shallow renders", () => {
    const wrapper = shallow(<HelloWorld />);
    const instance = wrapper.instance();
    console.log(instance);
    expect(instance.state.text).toEqual("Placeholder");
  });
});
