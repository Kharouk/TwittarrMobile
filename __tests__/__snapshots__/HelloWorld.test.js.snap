// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 40,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Send Message:
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    underlineColorAndroid="transparent"
    value="Placeholder"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Send
      </Text>
    </View>
  </View>
</View>
`;
